File: Exams/E1/cpp/E1.cpp
#include <vector>
#include <iostream>
using namespace std;
class Student
{
private:
    std:: vector <double> Grades;
    std:: vector <int> Weights;
public:
    Student(std::vector<double>G,std::vector<int>W)
    {
        Grades = G;
        Weights = W;
        
    }
    double GetAvg()
    {
        double sum = 0;
        int w = 0;
        for (int i = 0; i < Grades.size(); i++)
        {
            sum+=Grades[i]*Weights[i];
            w+=Weights[i];
        }
       return sum/w; 
    }
};
int main(int argc, char const *argv[])
{
    vector<double>Grades = {16,15.25,17,18};
    vector<int>Weights = {1,2,3,2};
    Student s(Grades,Weights);
    cout<<s.GetAvg();
    return 0;
}

File: Exams/E1/python/e1.py

class Student:
    Grades = []
    Weights = []
    def __init__(self,G,W):
        for i in range(len(G)):
            self.Grades.append(G[i])
            self.Weights.append(W[i])

    def GetAvg(self):
        sum = 0
        w = 0
        for i in range(len(self.Grades)):
        
            sum+=self.Grades[i]*self.Weights[i]
            w+=self.Weights[i]
        
        return sum/w

if __name__ == "__main__":
    Grades = [15,18,17,20,16,14]
    Weights = [2,3,3,1,2,1]
    s = Student(Grades,Weights)
    print(s.GetAvg())